akka-projection-core/src/main/scala/akka/projection/javadsl/SourceProvider.scala

```diff
@@ -5,12 +5,13 @@
 package akka.projection.javadsl
 
 import java.util.Optional
+import java.util.concurrent.CompletionStage
 
 import akka.stream.javadsl.Source
 
 trait SourceProvider[Offset, Envelope] {
 
-  def source(offset: Optional[Offset]): Source[Envelope, _]
+  def source(offset: () => CompletionStage[Optional[Offset]]): CompletionStage[Source[Envelope, _]]
```

Comment:
This will be a `scala.Function0` in the Java API, it should be a `Supplier`.

Reply:
Thank you.  I had a feeling that wasn't right.